server/src/test/java/io/crate/execution/engine/collect/CollectTaskTest.java

```diff
@@ -94,58 +115,10 @@ public void testAddingSameContextTwice() throws Exception {
             assertFalse(true); // second addContext call should have raised an exception
         } catch (IllegalArgumentException e) {
             verify(mock1, times(1)).close();
-            verify(mock2, times(1)).close();
+            verify(mock2, times(0)).close(); // would be closed via `kill` on the context
         }
     }
 
-    @Test
-    public void testInnerCloseClosesSearchContexts() throws Exception {
-        RefCountedItem mock1 = mock(RefCountedItem.class);
-        RefCountedItem mock2 = mock(RefCountedItem.class);
-
-        collectTask.addSearcher(1, mock1);
-        collectTask.addSearcher(2, mock2);
-
-        collectTask.innerClose();
-
-        verify(mock1, times(1)).close();
-        verify(mock2, times(1)).close();
-    }
-
-    @Test
-    public void testKillOnJobCollectContextPropagatesToCrateCollectors() throws Exception {
```

Comment:
Replaced this with multiple `test_kill_after_start` variants - see further below